oeps/architectural-decisions/oep-0065/decisions/0002-frontend-app-migrations.rst

```diff
@@ -0,0 +1,152 @@
+* Built and released as an NPM package for build-time inclusion in a frontend :term:`Project`, perhaps alongside other application modules from other libraries.
+
+Environment Agnostic
+====================
+
+The :term:`application module libraries <Application Module Library>` will no longer contain ``.env`` or ``env.config`` files for any specific environment, including `Devstack <https://github.com/openedx/devstack>`_ and `Tutor <https://docs.tutor.edly.io/>`_. Config filename patterns will be added to the ``.gitignore`` file. They will continue to support adding a (git ignored) config file into the repository to build or develop it, but we also expect operators to use Projects and check their config files into those project repositories as their primary way of working with the application module libraries.
+
+Please see :doc:`0003-frontend-projects` for more information on projects.
+
+Removed Dependencies
+====================
+
+Application module libraries will cease to use the following libraries in favor of ``frontend-base``:
+
+* @openedx/frontend-build
+* @edx/frontend-plaform
+* @openedx/frontend-plugin-framework
+* @edx/frontend-component-header
+* @edx/frontend-component-footer
+* @openedx/frontend-slot-footer
+* @edx/brand
+* core-js
+* regenerator-runtime
+
+Peer Dependencies
+=================
+
+We expect application module libraries to be dependencies of Frontend Projects by default for most operators. Because of this, the following dependencies will become peer dependencies in the application module libraries themselves:
+
+* @openedx/frontend-base
+* @openedx/paragon
+* react
+* react-dom
+* react-redux
+* react-router
+* react-router-dom
+* redux
+
+New CLI Tools
+=============
+
+The ``fedx-scripts`` CLI tools from ``frontend-build`` will be replaced with the ``openedx`` CLI tools from ``frontend-base``. We'll discuss some of them in detail here, as they help illustrate what the library will be able to do:
+
+* ``dev`` will start a dev server, loading the repository's application modules into the shell in a site.
+* ``dev:module`` will start a dev server that provides the application modules via module federation.
+* ``build`` will create a standalone deployable artifact that uses the shell (similar to the micro-frontend architecture)
+* ``build:module`` will create a standalone deployable artifact that provides the application modules via module federation.
+* ``release`` will package the library for distribution on npm.
+* ``serve`` will work with ``build`` or ``build:module`` to locally serve the production assets they generated.
+* ``pack`` will work with ``release`` to create a ``.tgz`` file suitable for installing in local git checkouts that depend on the library. (this is a development tool)
+
+The ``dev``, ``dev:module``, ``build``, and ``build:module`` CLI commands will rely on the existence of a :term:`Site Config` file (the replacement for .env/env.config files) which will not be checked into the repository.
```

Comment:
Not necessarily a suggestion, but a request for clarification:

```suggestion
The ``dev``, ``dev:module``, ``build``, and ``build:module`` CLI commands will rely on the existence of a :term:`Site Config` file (the replacement for .env/env.config files) which will not be checked into the module repository.
```

The config file will be checked into _a_ repository (the project's, I suppose), just not the module's... right?

Reply:
Good clarification, fixing.